### PR TITLE
Feature: add Travis CI

### LINKED ISSUE #156: Setup Travis CI
In order to be sure that every merged / committed change at least compiles and there are no library version conflicts.

### PR DESCRIPTION
**Pull request by aaabramov**
Fixes #156 

Also adds documentation for `qos` parameter

### CODE REVIEW

**Comment by aaabramov** [comment]
Very strange error message at https://travis-ci.org/SpinGo/op-rabbit/jobs/430658764:
```
Download failed. Obtain the jar manually and place it at /home/travis/.sbt/launchers/1.1.5/sbt-launch.jar
```

Has anyone faced that before?
The same error described here: https://gist.github.com/kshakir/3b0b915731444a03a331f6e2677a46aa#sbt_triple_download_attempt_with_not_foundtxt

**Comment by aaabramov** [comment]
Seems that Travis has no support for 2.11.12 on jdk7.
Nevertheless, builds are fixed for cross-published versions.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  - oraclejdk8
+
+# inspired by https://github.com/hashicorp/nomad-scala-sdk/blob/master/.travis.yml
+matrix:
+  exclude:
+  # We build Scala 2.11 on Java 7, so let's not bother with Java 8
+  - jdk: oraclejdk8
+    scala: 2.11.12
+  # Scala 2.12+ requires Java 8
+  - jdk: oraclejdk7
```
Comment (timcharper) [inline .travis.yml]: these seem backwards; why not use jdk8 for both?

Comment (aaabramov) [reply .travis.yml]: let me try

**Comment by aaabramov** [comment]
Non PR context question.

@timcharper do you plan to support this library further?
I am interesting because I would like to start using it in my project.

By supporting, I mean add new RabbitMQ features, etc.

File: .travis.yml (hunk 1 above)
Comment (aaabramov) [reply .travis.yml]: Seems that it should be fine: https://docs.scala-lang.org/overviews/jdk-compatibility/overview.html

Comment (aaabramov) [reply .travis.yml]: Done. Job is successful

**Comment by aaabramov** [comment]
@timcharper any concerns | blockers for this to be merged?

**Comment by timcharper** [comment]
@aaabramov none :) We had this on before and it failed really bad, and I was never sure while. Let's try it again!

**Pull request merged**